plugins/ltac/tacinterp.ml

```diff
@@ -2041,7 +2044,9 @@ let _ =
     let extra = TacStore.set TacStore.empty f_debug (get_debug ()) in
     let ist = { lfun = lfun; extra; } in
     let tac = interp_tactic ist tac in
-    let name, poly = Id.of_string "ltac_sub", false in
+    (* XXX: This depends on the global state which is bad; the hooking
+       mechanism should be modified. *)
+    let name, poly = Id.of_string "ltac_gen", false in
```

Comment:
Here is the remaining test-suite failure, `poly` is not propagated properly by the `ltac:` interpretation.

This is tricky to fix, as this is usually called from the interpretation of an argument `casted_constr` for the case of the test suite `exact ltac:(...)`, which in turn is hooked with `ArgInterpLegacy`.

In order to be not very invasive I've added a pretyping flags field.

Reply:
There is a possible fix at https://github.com/ejgallego/coq/tree/geninterp_poly but I dunno what @ppedrot thinks.

I wouldn't consider it a blocker, how to address the general issue seems like a bit orthogonal.